napari/utils/interactions.py

```diff
@@ -210,15 +210,37 @@ def hello_world(layer, event):
     'Enter': '↵',
 }
 
-
+joinchar = '-'
 if sys.platform.startswith('darwin'):
     KEY_SYMBOLS.update(
         {'Control': '⌘', 'Alt': '⌥', 'Option': '⌥', 'Meta': '⌃'}
     )
+    joinchar = ''
 elif sys.platform.startswith('linux'):
     KEY_SYMBOLS.update({'Meta': 'Super'})
 
 
+def format_shortcut(shortcut: str) -> str:
+    """
+    Format the given shortcut for the current platform.
+
+    Replace Cmd, Ctrl, Meta...etc by appropriate symbols if relevant for the
+    given platform.
+
+    Parameters
+    ----------
+    shortcut: string
+        shortcut to format in the form of dash separated keys to press
+
+    Returns
+    -------
+    string
+        Shortcut formatted to be displayed on current paltform.
+    """
+
+    return joinchar.join(KEY_SYMBOLS.get(x, x) for x in shortcut.split('-'))
```

Comment:
Another complication — aren't `Ctrl--` and `Ctrl-+` or `Ctrl+-` and `Ctrl++` valid shortcuts?

Reply:
That has been updated in the other PR. Yes it might be an issue for `Ctrl--`, but `Ctrl+...` is the QT form, so should not be present in this code base.

Now there is a Shortcut object (not a string) and so we can change the internal representation instead of storing it as a string. That should give us more flexibility.

Reply:
See https://github.com/napari/napari/pull/2554 @jni